generic/pg-user.el

```diff
@@ -597,6 +597,223 @@ last use time, to discourage saving these into the users database."
 
 
 
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;
+;; Check validity of proofs
+;;
+
+(defun proof-check-report (proof-results tap batch)
+  "Report `proof-check-proofs' results in PROOF-RESULTS in special buffer.
+Report the results of `proof-check-proofs' in buffer
+`proof-check-report-buffer' in human readable form or, if TAP is
+not nil, in test anything protocol (TAP). If BATCH is not nil,
+report the results via message, such that they appear on stdout
+when Emacs runs in batch mode or, when BATCH is a string, append
+the results to the file denoted by BATCH."
+  (let* ((ok-fail (seq-group-by #'car proof-results))
+         (frmt "  %-4s %s")
+         (frmt-face (propertize frmt 'face 'error))
+         (count 1)
+         coq-proj-dir src-file)
+
+    ;; determine a relative file name for current buffer
+    (when buffer-file-name
+      (setq coq-proj-dir (locate-dominating-file buffer-file-name
+                                                 coq-project-filename)))
+    (if coq-proj-dir
+        (setq src-file (file-relative-name buffer-file-name coq-proj-dir))
+      (setq src-file (buffer-name)))
+
+    ;; generate header
+    (with-current-buffer (get-buffer-create proof-check-report-buffer)
```

Comment:
Some feedback:
* (important) I believe the `*proof-check-report*` should be read-only, adding `(read-only-mode)` somewhere here
* (minor, maybe to do later, not in this PR!) It could be nice to just type `RET` on one of the lemma name, to directly switch to it (handy if the .v file is very large), you know, a bit like the `*ert*` view of `M-x ert`

Reply:
* read-only: good point
* key binding: good idea, but I am not motivated to do this now. Instead I envision an extension of `proof-check-proofs` that annotates failing theorems in the Coq sources with `(* FAIL *)`. For my repos I then want to set up a CI action that only checks the consistency of these `FAIL` comments using `proof-check-proofs`. This way, fixed proofs and new failing ones pop up prominently during review and one can search form them. Moreover, `M-.` (`xref-find-definitions`) works in the report buffer.

Reply:
> annotates failing theorems in the Coq sources with `(* FAIL *)`

OK, looks like a nice idea!

> `M-.` works in the report buffer.

Great!